code/game/machinery/portable_turret.dm

```diff
@@ -733,35 +622,30 @@ Status: []<BR>"},
 	if (!raised) // the turret has to be raised in order to fire - makes sense, right?
 		return
 
-
-	// any emagged turrets will shoot extremely fast! This not only is deadly, but drains a lot power!
-
-	if(iconholder)
+	if(!emagged)
 		icon_state = "[lasercolor]target_prism"
 	else
 		icon_state = "[lasercolor]orange_target_prism"
 
+	use_power(reqpower)
+
+//		 //Shooting Code:
+	playsound(get_turf(src), installed.fire_sound, 75, 1)
 	var/obj/item/projectile/A
-	if(emagged)
-		playsound(get_turf(src), efire_sound, 75, 1)
-		A = new eprojectile( loc )
-	else
-		playsound(get_turf(src), fire_sound, 75, 1)
-		A = new projectile( loc )
-	A.original = target.loc
-	if(!emagged)
-		use_power(reqpower)
-	else
-		use_power((reqpower*2))
-		// Shooting Code:
+	A = new installed.projectile_type(loc)
+	A.original = target
 	A.starting = T
 	A.shot_from = src
 	A.current = T
 	A.yo = U.y - T.y
 	A.xo = U.x - T.x
-	spawn( 1 )
-		A.OnFired()
-		A.process()
+	A.OnFired()
```

Comment:
Why have you unspawned these? To my knowledge, neither are waitfor = 0 and will delay the proc.